tests/50federation/30room-join.pl

```diff
@@ -52,103 +52,118 @@ sub assert_is_valid_pdu {
 }
 push our @EXPORT, qw( assert_is_valid_pdu );
 
-test "Outbound federation can send room-join requests",
-   requires => [ local_user_fixture(), $main::INBOUND_SERVER,
-                 federation_user_id_fixture() ],
+foreach my $versionprefix ( qw( v1 v2 ) ) {
+   test "Inbound federation supports $versionprefix /send_join",
+      requires => [ local_user_fixture(), $main::INBOUND_SERVER,
+                    federation_user_id_fixture() ],
 
-   do => sub {
-      my ( $user, $inbound_server, $creator_id ) = @_;
+      do => sub {
+         my ( $user, $inbound_server, $creator_id ) = @_;
 
-      my $local_server_name = $inbound_server->server_name;
-      my $datastore         = $inbound_server->datastore;
+         my $local_server_name = $inbound_server->server_name;
+         my $datastore         = $inbound_server->datastore;
 
-      my $room = SyTest::Federation::Room->new(
-         datastore => $datastore,
-      );
+         my $room = SyTest::Federation::Room->new(
+            datastore => $datastore,
+         );
 
-      $room->create_initial_events(
-         server  => $inbound_server,
-         creator => $creator_id,
-      );
+         $room->create_initial_events(
+            server  => $inbound_server,
+            creator => $creator_id,
+         );
 
-      my $room_id = $room->room_id;
+         my $room_id = $room->room_id;
 
-      my $room_alias = "#50fed-room-alias:$local_server_name";
-      $datastore->{room_aliases}{$room_alias} = $room_id;
+         my $room_alias = "#50fed-room-alias:$local_server_name";
+         $datastore->{room_aliases}{$room_alias} = $room_id;
 
-      Future->needs_all(
-         # Await PDU?
+         my $await_request_send_join;
 
-         $inbound_server->await_request_make_join( $room_id, $user->user_id )->then( sub {
-            my ( $req, $room_id, $user_id ) = @_;
+         if( $versionprefix eq "v1" ) {
+            $await_request_send_join = $inbound_server->await_request_v1_send_join( $room_id );
+         }
```

Comment:
surely in the v1 case we should expect that the v2 request happens first; we can then 404 it and wait for the v1 request

Reply:
Why? SyTest already 404s requests on the endpoints it's not expecting requests on.

Reply:
doesn't it spit out some warnings about it?

Reply:
Nope:

```
$ ./run-tests.pl tests/50federation/30room-join.pl
Running tests/50federation/30room-join.pl...
  Testing if: Outbound federation can query v1 /send_join... 
Tests: 0 / 17 | OKSignature ok
subject=CN = localhost
Getting CA Private Key
Signature ok
subject=CN = localhost
 # Started test HTTPS Server at localhost:42521
 # Started test SMTP Server at localhost:34521
 # Starting Homeserver using SyTest::HomeserverFactory::Synapse=HASH(0x556a57bad378)
 # Starting server-0
 # Clearing SQLite database at :memory:
  Testing if: Outbound federation can query v1 /send_join... 
Tests: 0 / 17 | OKSignature ok
subject=CN = localhost
 # Generating config for port 8800
 # Creating config for server 0 with command python -m synapse.app.homeserver --config-path /home/brendan/Documents/matrix/sytest-stable/server-0/config.yaml --server-name localhost:8800 --generate-config --report-stats=no
  Testing if: Outbound federation can query v1 /send_join... 
 # Starting server 0 for port 8800 with command python -m synapse.app.homeserver --config-path /home/brendan/Documents/matrix/sytest-stable/server-0/config.yaml --server-name localhost:8800
 # Connecting to server 8800
 # Connected to server 8800
 # Started server-0
  Testing if: Outbound federation can query v1 /send_join... PASS
  Testing if: Outbound federation can query v2 /send_join... PASS
  Testing if: Outbound federation passes make_join failures through to the client... PASS
  Testing if: Inbound federation can receive v1 /send_join... PASS
  Testing if: Inbound federation can receive v2 /send_join... PASS
...
```

Reply:
fair enough. can you chuck in a comment that you're relying on that behaviour?

Reply:
Done in https://github.com/matrix-org/sytest/pull/760/commits/5db16de9fda73c1558565cc2f5181cfde10cd7a9.